src/classes/RD2_RecurringDonations_TEST.cls

```diff
@@ -332,34 +334,66 @@ private with sharing class RD2_RecurringDonations_TEST {
     }
 
     /***
-    * @description Verifies that start date is updated to current date on updated RD when appropriate
+    * @description Verifies that start date is NOT updated to current date on updated RD when appropriate
     */
     @isTest
-    private static void shouldOverrideStartDateWithCurrentDate() {
-        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
-
-        Contact contact = getContact();
-
-        npe03__Recurring_Donation__c rd = TEST_RecurringDonationBuilder.constructEnhancedBuilder()
-            .withContact(contact.Id)
-            .withAmount(50)
-            .withInstallmentPeriod(RD2_Constants.INSTALLMENT_PERIOD_MONTHLY)
-            .withDayOfMonth('10')
-            .withInstallmentFrequency(1)
-            .withStatus(RD2_Constants.STATUS_ACTIVE)
-            .withDateEstablished(Date.newInstance(2019, 6, 5))
-            .withStartDate(Date.newInstance(2019, 6, 5))
+    private static void shouldNotOverrideStartDateWithCurrentDate() {
+        Date startDate = Date.newInstance(2019, 10, 5);
+        npe03__Recurring_Donation__c rd = getRecurringDonationMonthlyBuilder()
+            .withStartDate(startDate)
             .build();
 
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
         Test.startTest();
         insert rd;
-        rd.npe03__Amount__c = 100;
+        rd.npe03__Amount__c = 250;
         rd.InstallmentFrequency__c = 2;
         update rd;
         Test.stopTest();
 
+        List<RecurringDonationSchedule__c> rdSchedules = rdScheduleGateway.getRecords(rd);
+        System.assertEquals(2, rdSchedules.size(), 'Should be 2 schedules.');
+
+        System.assertEquals(startDate, rdSchedules[0].StartDate__c, 'Original start date should be 10/5/2019.');
```

Comment:
Since we would like to address and document expected behaviour in tests, mentioning dates in assert statements can bring extra detail that might be redundant. Based on the assert statements, it seems that expected behaviour is to neutralize the existing original schedule and create a new schedule valid from the original schedule Start Date? 

If so, then we can write tests to document behaviour:
```
System.assertEquals(startDate, rdSchedules[0].StartDate__c, 'Original schedule Start Date should be unchanged');
System.assertEquals(startDate.addDays(-1), rdSchedules[0].EndDate__c, 'Original schedule End Date should be changed so the schedule is invalid');
System.assertEquals(startDate, rdSchedules[1].StartDate__c, 'New schedule Start Date should be original schedule Start Date ');
System.assertEquals(null, rdSchedules[1].EndDate__c, 'New schedule End Date should be undefined');
```

Reply:
@force2b ^, is the expected behaviour for the old schedule to be set as invalid when Amount changes and Effective Date (which is in past) is unchanged?

Reply:
I believe that we previously determined any changes to the contents of the schedule (dates, amounts, payment methods, etc.) will cause the existing Schedule record to be set as inactive (gets an end date) and a new schedule record is created. However, I think what you're asking is if we have a Schedule record that hasn't really started yet and a change is made to a field on the RD should the  Schedule record just be updated rather than create a new one.

I suppose that technically either are valid in this scenario, however I would choose the path of least resistance in terms of code/logic changes. If the existing behavior works fine - where it closes the previous schedule and creates a new one - then I'm not sure there's a lot of value in refactoring it to do something different in a unique case such as this.

Reply:
@RadSFDeveloper @force2b Our logic so far is that every schedule change puts an end date on any old schedule and creates a new schedule with an appropriate start date (sometimes the end date on a schedule is before the start date if we consider the schedule to be permanently disabled).  If schedules were reportable user records we might not approach it this way, but given the fact that schedules are permanently and irrevocably inaccessible to our users, the only thing we should value is the clarity of what a schedule is intended to do, and this is to drive the creation of future opportunities.  For that, it doesn't matter at all whether we update or insert a new schedule, so I don't see any value to the increased complexity that results from doing thing in two different ways.

Reply:
Thank you for the clarification!